sublime-text/SublimeKite.py

```diff
@@ -179,25 +200,45 @@ def _update(self, action, view):
             'pluginId': self.PLUGIN_ID,
         })
 
-        if PYTHON_VERSION >= 3:
-            json_body = bytes(json_body, "utf-8")
-
-        self._write_sock(json_body)
-
-    def _error(self, data):
+    def _error(self, msg):
         view = sublime.active_window().active_view()
-        json_body = json.dumps({
+        self._http_roundtrip(ERROR_ENDPOINT, {
             'source': SOURCE,
-            'action': "error",
             'filename': realpath(view.file_name()),
-            'text': json.dumps(data),
-            'pluginId': self.PLUGIN_ID,
+            'message': msg,
         })
+        print(msg)
+
+    def _http_roundtrip(self, endpoint, payload):
+        """
+        Send a json payload to kited at the specified endpoint
+        """
+        resp = None
+        try:
+            verbose("sending to", endpoint, ":", payload)
+            req = json.dumps(payload)
+
+            if PYTHON_VERSION >= 3:
+                import http.client
+                conn = http.client.HTTPConnection(KITED_HOSTPORT)
```

Comment:
Should we bother trying to reuse connections at all? If not now, probably in the future.